src/encoded/static/components/matrix_deeply_profiled.js

```diff
@@ -0,0 +1,576 @@
+        const prospectiveMax = Math.max(...rowDataValues);
+        if (maxSubCategoryValue < prospectiveMax) {
+            maxSubCategoryValue = prospectiveMax;
+        }
+        const prospectiveMin = Math.min(...rowDataValues.filter((value) => value));
+        if (minSubCategoryValue > prospectiveMin) {
+            minSubCategoryValue = prospectiveMin;
+        }
+    });
+    return { subCategorySums, maxSubCategoryValue, minSubCategoryValue: minSubCategoryValue - 1 };
+};
+
+
+let _navbarHeight = null;
+
+/**
+ * Get height of the navbar
+ *
+ * @returns height of navbar on first call and restores same value until page is garbage collected
+ */
+const getNavbarHeight = () => {
+    if (_navbarHeight === null) {
+        const navbar = document.querySelector('#navbar');
+        _navbarHeight = navbar ? navbar.getBoundingClientRect().height : 0;
+    }
+    return _navbarHeight;
+};
+
+/**
+ * Takes matrix data from JSON and generates an object that <DataTable> can use to generate the JSX
+ * for the matrix. This is a shim between the experiment or audit data and the data <DataTable>
+ * needs.
+ * @param {object} context Matrix JSON for the page
+ * @param {func}   getRowCategories Returns rowCategory info including desired display order
+ * @param {func}   getRowSubCategories Returns subCategory desired display order
+ * @param {func}   mapRowCategoryQueries Callback to map row category query values
+ * @param {func}   mapSubCategoryQueries Callback to map subcategory query values
+ * @param {array}  expandedRowCategories Names of rowCategories the user has expanded
+ * @param {func}   expanderClickHandler Called when the user expands/collapses a row category
+ *
+ * @return {object} Generated object suitable for passing to <DataTable>
+ */
+
+const convertExperimentToDataTable = (context, getRowCategories, getRowSubCategories, mapRowCategoryQueries, mapSubCategoryQueries, expandedRowCategories, expanderClickHandler) => {
+    const rowCategory = context.matrix.y.group_by[0];
+    const subCategory = context.matrix.y.group_by[1];
+    const columnCategoryType = context.matrix.x.group_by;
+
+    // Generate the top-row sideways header labels. First item is null for the empty upper-left
+    // cell.
+    const colCount = context.matrix.x[context.matrix.x.group_by].buckets.length;
+    const colTitleMap = {};
+    const colCategoryNames = context.matrix.x[context.matrix.x.group_by].buckets.map((colCategoryBucket, colIndex) => {
+        colTitleMap[colCategoryBucket.key] = colIndex;
+        return colCategoryBucket.key;
+    });
+
+    // Set specific base urls, in different combinations
+    let query = new QueryString(context.search_base);
+    query.deleteKeyValue(subCategory);
+    const baseUrlWithoutSubcategoryType = query.format();
+
+    query.deleteKeyValue(columnCategoryType);
+    const baseUrlWithoutSubNorColCategoriesType = query.format();
+
+    query = new QueryString(context.search_base);
```

Comment:
You can also `query.clone()` as soon as you create it above, in which it does a `new QueryString`.